services/inc/enumclass.h

```diff
@@ -22,12 +22,225 @@
 
 namespace particle {
 
-template <typename E>
-constexpr auto to_underlying(E e) noexcept
-{
+template <typename T, typename Enable = void>
+class BitMaskFlags;
+
+template<typename T>
+struct EnableBitwise {
+    static const bool enable = false;
+};
+
+#define ENABLE_ENUM_CLASS_BITWISE(T) \
+template<>                           \
+struct EnableBitwise<T> {            \
+    static const bool enable = true; \
+}
+
+template<typename T>
+class BitMaskFlags<T, typename std::enable_if_t<std::is_enum<T>::value>> {
```

Comment:
Let's move this class into its own header named after the class, insead of `enumclass.h`. Let's also perhaps rename to `EnumFlags`, as that tells the intended usage better.